### PR TITLE
StlPlotter now saves a png instead of displaying to screen.

### LINKED ISSUE #4: Add ability to save plot as PNG to StlPlotter
Calling `StlPlotter#plot` causes Travis CI to fail because X11 is not set up. PlotTest has been disabled for now to allow CI to pass, but we should add the ability to save plot as PNG as opposed to rendering on screen, so we can continue to exercise code path.

### PR DESCRIPTION
**Pull request by hntd187**
StlPlotter now saves the file instead of displaying it to the screen for travisCI. Also enabled the test for the plotter and added some new tests to make sure the plot is generated. This should fix #4 , better late than never right? :zzz: :zzz:

### CODE REVIEW

File: src/test/java/com/github/brandtg/PlotTest.java [hunk 1]
```diff
@@ -35,6 +36,17 @@ public void testPlot() throws Exception {
...
     final StlResult res = stl.decompose(ts, ys);
+
+    final File output = new File("seasonal.png");
+    final File hourly = new File("stl-hourly.png");
+
+    StlPlotter.plot(res, "New Title", Hour.class, hourly);
+    StlPlotter.plot(res, output);
     StlPlotter.plot(res);
+
+    assert (hourly.exists());
```
Comment (brandtg) [inline src/test/java/com/github/brandtg/PlotTest.java]: Can you change `assert` to TestNG's `Assert.assertTrue(...)` http://testng.org/javadocs/org/testng/Assert.html#assertTrue-boolean-

**Comment by brandtg** [comment]
No worries, thanks for your contribution! Just a minor comment w/ the test framework usage

File: src/test/java/com/github/brandtg/PlotTest.java [hunk 2]
```diff
@@ -35,6 +38,17 @@ public void testPlot() throws Exception {
 
     final StlDecomposition stl = new StlDecomposition(config);
     final StlResult res = stl.decompose(ts, ys);
+
+    final File output = new File("seasonal.png");
+    final File hourly = new File("stl-hourly.png");
+
+    StlPlotter.plot(res, "New Title", Hour.class, hourly);
+    StlPlotter.plot(res, output);
     StlPlotter.plot(res);
```
Comment (brandtg) [inline src/test/java/com/github/brandtg/PlotTest.java:60]: I think we need to remove the `StlPlotter.plot` calls that do not output to a file, as the travis build fails with the following error still

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running TestSuite
Configuring TestNG with: TestNG652Configurator
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.897 sec <<< FAILURE! - in TestSuite
testPlot(com.github.brandtg.PlotTest)  Time elapsed: 3.015 sec  <<< FAILURE!
java.awt.HeadlessException: 
No X11 DISPLAY variable was set, but this program performed an operation which requires it.
    at java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:207)
    at java.awt.Window.<init>(Window.java:536)
    at java.awt.Frame.<init>(Frame.java:420)
    at javax.swing.JFrame.<init>(JFrame.java:225)
    at org.jfree.ui.ApplicationFrame.<init>(ApplicationFrame.java:65)
    at com.github.brandtg.StlPlotter$ResultsPlot.<init>(StlPlotter.java:61)
    at com.github.brandtg.StlPlotter.plot(StlPlotter.java:42)
    at com.github.brandtg.PlotTest.testPlot(PlotTest.java:45)
```

**Comment by hntd187** [comment]
Bah okay @brandtg why don't we just turn this off? I think jfreechart does an X11 call no matter what. I suppose these tools should just exist as user tools and not something we can throw in the CI loop.

**Comment by brandtg** [comment]
@hntd187 Fair enough, SGTM. Regarding testing though, I do think we could benefit from something that generates some series with known trend + seasonal and noise / anomalies, then see if we can decompose and get something similar w/in error bounds. Something similar to this but maybe less simplistic: https://github.com/brandtg/stl-java/blob/master/src/test/java/com/github/brandtg/TestStlDecomposition.java#L42

**Comment by hntd187** [comment]
@brandtg ballin out of control, that should let us continue to test this and keep travis happy. Let's open a new issue for the other suggestion you made.

**Comment by brandtg** [comment]
SGTM

**Pull request merged**